hydro_lang/src/ir.rs

```diff
@@ -28,6 +28,47 @@ use crate::backtrace::Backtrace;
 use crate::deploy::{Deploy, RegisterPort};
 use crate::location::LocationId;
 
+/// Represents the kind of stream/collection type for metadata
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+pub enum StreamKind {
+    /// Regular stream - ordering/retries determined by type parameters
```

Comment:
What does "ordering/retries determined by type parameters" mean? In the metadata those type parameters are not available so doesn't make sense. Probably should just be `/// Regular stream`